src/Go/SaveLoad.ts

```diff
@@ -88,14 +90,18 @@ export function loadGo(data: unknown): boolean {
       showError(new Error(`Error while making first IPvGO AI move: ${error}`, { cause: error }));
     });
   }
-  // If it's not the AI's turn and we're not in gameover status, initialize nextTurn promise based on the previous move/pass
+  // If it's not the AI's turn (and we're not in gameover status), initialize nextTurn promise based on the previous move/pass
   else if (currentGame.previousPlayer) {
     const previousMove = getPreviousMove();
     Go.nextTurn = Promise.resolve(
       previousMove
         ? { type: GoPlayType.move, x: previousMove[0], y: previousMove[1] }
         : { type: GoPlayType.pass, x: null, y: null },
     );
+    Go.nextTurnForWhite = cloneDeep(Go.nextTurn);
```

Comment:
This is deeply suspicious to me. What happens if the ai is *not* GoOpponent.none?

Also, why even clone this? What's the issue with them referencing the same promise, if this usage was OK in the first place?

Reply:
This was part of testing for a bug where the wrong promise was being resolved, but I don't need it anymore. Good catch